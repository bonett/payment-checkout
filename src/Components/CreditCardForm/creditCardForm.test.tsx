import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import CreditCardComponent from "./";
import store from "Store";

describe("CreditCard component", () => {
  test("Should renders CreditCard component", () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CreditCardComponent/>
        </BrowserRouter>
      </Provider>
    );

    const numberLabel = screen.getByText(/Enter card number/i);
    const nameLabel = screen.getByText(/Enter holdname/i);
    const expiryLabel = screen.getByText(/Enter expiry date/i);
    const cvcLabel = screen.getByText(/Enter cvc code/i);
    expect(numberLabel).toBeInTheDocument();
    expect(nameLabel).toBeInTheDocument();
    expect(expiryLabel).toBeInTheDocument();
    expect(cvcLabel).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
