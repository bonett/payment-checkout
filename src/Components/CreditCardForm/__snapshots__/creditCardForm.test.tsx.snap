// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCard component Should renders CreditCard component 1`] = `
<div>
  <form
    class="form-content"
  >
    <div
      class="form-content_naming"
    >
      <label>
        Enter card number
      </label>
      <input
        maxlength="16"
        name="number"
        placeholder="Card number"
        type="tel"
        value=""
      />
    </div>
    <div
      class="form-content_naming"
    >
      <label>
        Enter holdname
      </label>
      <input
        maxlength="20"
        name="name"
        placeholder="Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-content_naming"
    >
      <label>
        Enter expiry date
      </label>
      <input
        maxlength="5"
        name="expiry"
        placeholder="MM/YY"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-content_naming"
    >
      <label>
        Enter cvc code
      </label>
      <input
        maxlength="3"
        name="cvc"
        placeholder="CVC"
        type="tel"
        value=""
      />
    </div>
  </form>
</div>
`;
